apisix/plugins/hmac-auth.lua

```diff
@@ -246,10 +246,13 @@ local function validate(ctx, params)
 
     core.log.info("clock_skew: ", conf.clock_skew)
     if conf.clock_skew and conf.clock_skew > 0 then
-        local diff = abs(ngx_time() - params.timestamp)
-        core.log.info("timestamp diff: ", diff)
-        if diff > conf.clock_skew then
-          return nil, {message = "Invalid timestamp"}
+        local time = ngx.parse_http_time(params.date)
```

Comment:
The value obtained here may be empty, we need to perform error handling,  right?

Reply:
it hasn't a err here. if params.date is empty string or other invalid string, it will return nil.

Reply:
if the `time` is nil, I think we should throw an error message and write an error log

Reply:
we allow it to be empty here

Reply:
This makes it easy to copy requests, bypassing the `clock_skew` option.
I don't think it can be empty here.

Reply:
OK.